src/node_file.cc

```diff
@@ -566,17 +569,21 @@ static void InternalModuleReadFile(const FunctionCallbackInfo<Value>& args) {
   CHECK_EQ(0, uv_fs_close(loop, &close_req, fd, nullptr));
   uv_fs_req_cleanup(&close_req);
 
-  size_t start = 0;
-  if (chars.size() >= 3 && 0 == memcmp(&chars[0], "\xEF\xBB\xBF", 3)) {
-    start = 3;  // Skip UTF-8 BOM.
-  }
+  if (numchars < 0) {
+    args.GetReturnValue().Set(Undefined(env->isolate()));
```

Comment:
+1 works for me.

On Sunday, September 4, 2016, Anna Henningsen <notifications@github.com>
wrote:

> In src/node_file.cc
> <https://github.com/nodejs/node/pull/8277#discussion_r77449453>:
>
> > @@ -566,17 +569,21 @@ static void InternalModuleReadFile(const FunctionCallbackInfo<Value>& args) {
> >    CHECK_EQ(0, uv_fs_close(loop, &close_req, fd, nullptr));
> >    uv_fs_req_cleanup(&close_req);
> >
> > -  size_t start = 0;
> > -  if (chars.size() >= 3 && 0 == memcmp(&chars[0], "\xEF\xBB\xBF", 3)) {
> > -    start = 3;  // Skip UTF-8 BOM.
> > -  }
> > +  if (numchars < 0) {
> > +    args.GetReturnValue().Set(Undefined(env->isolate()));
>
> See also the comment above the function:
>
> // Used to speed up module loading.  Returns the contents of the file as// a string or undefined when the file cannot be opened.  The speedup// comes from not creating Error objects on failure.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/nodejs/node/pull/8277/files/39db98ca5e20b1c28b5058c7b049747324d955bd#r77449453>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAa2eWcj_tAWKN-vBdhw8Czm5djPYWULks5qmpA1gaJpZM4Jt1xz>
> .
>